### DISCUSSION TITLE
Add a <forward_direction> member in <Node3D>

### DISCUSSION THREAD

**Discussion by jiaxiongjiao** [godotengine/godot-proposals#6053] [3D]
Hi, for the Node3D type, can we have some kind of <forward_direction> variable that uses a normalized Vector3D value to represent +Z vector value of a Node3D object in world space? So basically the same concept as `Transform.forward` in Unity.
https://docs.unity3d.com/ScriptReference/Transform-forward.html

**Comment by AThousandShips** [comment]
This would just be sugar for using `global_transform.basis * Vector3.FORWARD`, also note that forward is -Z in Godot, the coordinate system is different from Unity

You can also get this from `global_transform.basis.z`, which gets +Z

Correction, realized you need to transform by the basis or you will get the translation

**Reply by jon-heard** [reply]
This would be a good sugar to add, as the logic it is wrapping is not obvious to new Godot developers until after searching for a solution. Even after seeing it, many developers would likely forget the logic and need to re-search for the solution the first few times they need it.

**Comment by timothyqiu** [comment]
There's already a proposal for this: #5074
